src/community/spinglass/clustertool.cpp

```diff
@@ -256,6 +244,9 @@ static igraph_error_t igraph_i_community_spinglass_orig(
             IGRAPH_ERROR("Invalid weight vector length", IGRAPH_EINVAL);
         }
         use_weights = 1;
+        if (igraph_vector_min(weights) < 0) {
+            IGRAPH_ERROR("Cannot work with negative weights in this implementation.", IGRAPH_EINVAL);
```

Comment:
Let's be a bit more informative. Users of high-level interface may not realize what "this implementation" refers to, as they are likely not selecting the implementation explicitly.

```suggestion
            IGRAPH_ERROR("Weights must not be negative when using the original implementation of spinglass communities. Select the implementation meant for negative weights.", IGRAPH_EINVAL);
```